smoke-testing/src/test/java/games/strategy/engine/data/TestDataFileLister.java

```diff
@@ -0,0 +1,35 @@
+package games.strategy.engine.data;
+
+import java.io.File;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.stream.Collectors;
+import lombok.experimental.UtilityClass;
+import org.triplea.java.Postconditions;
+
+@UtilityClass
+class TestDataFileLister {
+
+  /**
+   * Returns a list of files located in folder like: 'test/resources/{folder-name}'.
+   *
+   * @param folderName Name of the folder in 'test/resources'
+   * @return List of all files located in specified folder.
+   */
+  Collection<File> listFilesInTestResourcesDirectory(final String folderName) {
+    return Arrays.stream(findFilesInFolder(folderName).listFiles())
+        .sorted(Comparator.comparing(File::getName))
+        .collect(Collectors.toList());
+  }
```

Comment:
Might be overkill, but you might want to consider creating a custom [ArgumentsProvider](https://junit.org/junit5/docs/current/api/org.junit.jupiter.params/org/junit/jupiter/params/provider/ArgumentsProvider.html).
If you combine this with a meta annotation like this:
```java
@Target({ANNOTATION_TYPE,METHOD})
@Retention(RUNTIME)
@ArgumentsSource(CustomArgumentSourceThatFindsDirectories.class)
public @interface DirectorySource {
  String folderName;
}
```

The `ArgumentsProvider` could then read out this annotation and provide the folders based on the name.
In my opinion a nice use of the API JUnit provides, but as mentioned already potentially overkill for just 2 methods.
Would avoid the repeated method source code though

Reply:
Interesting idea. I was not aware of the custom data source.
IMO we'll probably want around 3 to 5 examples before that would really makes sense.

Reply:
Yeah, at this point it's probably too much boilerplate to be worth it.
The JUnit API is really well designed in my opinion. It's a little bit heavy on reflection use in my opinion, but a lot of the code is designed around annotation inheritance, which makes it very extensible and customizable.
But there's definitely a hurdle to get the hang of it by observing how the junit implements csvsource for example